### ISSUE TITLE
Support uploading from file system (Node.js)

### ISSUE THREAD

**Issue by olalonde** [filestack/filestack-js#54]
I am writing tests that require me to upload a file to filestack via Node.js. It would be useful if the upload method would optionally accept a path.

**Comment by olalonde** [comment]
Related: https://github.com/filestack/filestack-js/issues/39

**Comment by velveteer** [comment]
@olalonde I have this feature implemented and tested, but no ETA on release yet. I'll update you with more information when I have it.

I don't know if it fits your use case, but [brfs](https://github.com/browserify/brfs) can help with getting filesystem assets into browser tests.

**Comment by juhaelee** [comment]
is it possible to upload files to filestack using node.js right now?

**Comment by rpavez** [comment]
Any ETA?

**Comment by velveteer** [comment]
We're going to be releasing `1.0.0-rc1` soon which will include ES and CommonJS modules for Node runtimes. I can't give a specific date yet, but very likely sometime in early March.

**Comment by rpavez** [comment]
Thats awesome, thanks!

**Comment by chornbe** [comment]
Seems the Filestack API was... lazily... written, and node doesn't natively support the convenience objects found in the browser implementations of JS. (sigh)

Shortcut after shortcut after shortcut. Kinda ready to kick Filestack to the curb.

**Comment by chornbe** [comment]
To clarify, using what *SEEMS* like the nice and simple code to upload from a Node module to Filestack via the filestack-js node api.

According to: https://www.npmjs.com/package/filestack-js#module_filestack..init.upload, saying "file Blob | string - must be a valid File, Blob, or base64 encoded string", I'm assuming the built-in string to b64 functions are sufficient...? 

I'm doing something like (forgive omissions, I chopped out all the existing biz logic, repeated and superfluous safety netting for this test, logging messages, etc.):

```
var filestack 		= require("filestack-js");
var client 		= filestack.init( config.fstoken );
module.exports = function( req, res ){
	form = new multiparty.Form();
	form.parse(req, function(err, fields, files) {
		console.log( "PARSE FORM DATA" );
		var file = files["upfile"][0];
		var path = file.path;
		fs.readFile( path, function( err, data ){
			// save to base64 encoded string
			var b64 = data.toString("base64");
			client.upload(
				b64,
				{ timeout: 1200000 }					
			)
			.then( function( res ){
				console.log( "UPLOAD DONE" );
				fs.unlinkSync( path );
				res.status(200).send( "done...?" );
				return;
			})
			.catch( function( err ){
				console.log( err );
				res.status(200).send( "error...?" );
				return;
			});
		});
	});
});

```
The error is really concerning...
## **ReferenceError: atob is not defined**

I guess what I'm asking is, the docs say that b64 data is ok to upload to filestack, and yet atob isn't present...? Am I cocking something up horribly? Did I misunderstand the docs? 

and the output...
```
PARSE FORM DATA
Bytes up: 4341756
/Users/chris/code/eyeneedServer/node_modules/filestack-js/dist/filestack.js:2
(function(global,factory){typeof exports==="object"&&typeof module!=="undefined"?module.exports=factory():typeof define==="function"&&define.amd?define(factory):global.filestack=factory()})(this,function(){"use strict";var DEFAULT_CNAME="filestackapi.com";var PICKER_VERSION="0.12.6";var envGetter=function envGetter(cname){return{picker:"https://static."+(cname||DEFAULT_CNAME)+"/picker/v3/picker-"+PICKER_VERSION+".js"}};var ENV={storeApiUrl:"https://www.filestackapi.com/api/store",fileApiUrl:"https://www.filestackapi.com/api/file",uploadApiUrl:"https://upload.filestackapi.com",cloudApiUrl:"https://cloud.filestackapi.com",processApiUrl:"https://process.filestackapi.com",previewApiUrl:"https://cdn.filestackcontent.com"};var isFunction=function isFunction(x){return typeof x==="function"};var isNil=function isNil(x){return x===null||x===void 0};var fail=function fail(message){throw new TypeError("[tcomb] "+message)};var getFunctionName
ReferenceError: atob is not defined
at b64toBlob (/Users/chris/code/eyeneedServer/node_modules/filestack-js/dist/filestack.js:2:129993)
at upload (/Users/chris/code/eyeneedServer/node_modules/filestack-js/dist/filestack.js:2:142319)
at Object.upload$$1 [as upload] (/Users/chris/code/eyeneedServer/node_modules/filestack-js/dist/filestack.js:2:145868)
at Object.upload (/Users/chris/code/eyeneedServer/node_modules/filestack-js/dist/filestack.js:2:150149)
at /Users/chris/code/eyeneedServer/routes/event/uploadPic.js:26:11
at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:528:3)

```

**Comment by jessica-pixvana** [comment]
I'm also experiencing this. Is there any workaround in the meantime?

**Comment by chornbe** [comment]
I never got it to work and went at S3 directly using the AWS api.

**Comment by filipenevola** [comment]
Hello @velveteer we are using filestack in my company and we need to make a decision to use it also in the back-end or send it directly to AWS S3 back-end uploads.

For now, as I can see, filestack doesn't have support for back-end upload. Is that coming? if yes, when? We really need that answer and would be good if you could provide it for us.

Thanks.

**Comment by velveteer** [comment]
Hey, so the `upload` method has an implementation for Node runtimes that will be available in 1.0.0-rc1. This method is the implementation of the multi-part flow, but uploads through the Filestack REST API are still an option for backends with HTTP clients: https://www.filestack.com/docs/rest-api/store.

We are aiming to publish 1.0.0-rc1 this week, but we're still doing a little QA on our end. I know I mentioned March in my last comment, but our scope was unexpectedly changed after making that post. We are working hard to make sure the next release is ready, and I thank everyone for their patience.

**Comment by velveteer** [comment]
The upload method for Node is now available in 1.0.0-rc1. Please see the release notes here: https://github.com/filestack/filestack-js/issues/101. 

If you feel this issue should be reopened please let us know.
